pkgs/development/python-modules/pysqlite3/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib
+, buildPythonPackage
+, pythonImportsCheckHook
+, fetchFromGitHub
+, fetchurl
+, python
+, sqlite
+, tcl
+, unittestCheckHook
+}:
+
+buildPythonPackage rec {
+  pname = "pysqlite3";
+  version = "0.5.2";
+
+  src = fetchFromGitHub {
+    owner = "coleifer";
+    repo = "pysqlite3";
+    rev = version;
+    hash = "sha256-Ik//afKc7v1iBRGwOZrQbbMsHmdH5FptS9EAldhKRmk=";
+  };
+
+  sqlitesrc = fetchurl {
+    url = "https://www.sqlite.org/src/tarball/sqlite.tar.gz";
+    hash = "sha256-jOGugJV9cD88/D3GH+uRJPm5V7LFbZIiubyOV/e4auo=";
+  };
+
+  preConfigure = ''
+    tar xzf $sqlitesrc
+    cd sqlite/
+    ./configure
+    make sqlite3.c
+    cp sqlite3.[ch] ../
+    cd ../
+  '';
```

Comment:
How can we extract source files from a `pkgs.sqlite.src`?

Reply:
Well, you can pass `sqlite3_src = sqlite.src` to `mkDerivation`, and your phases will have `sqlite3_src` environment variable which will point to the sqlite3 source, either directory or tarball, depending on whether sqlite3 uses fetchGit or fetchTarball.

This is not vendoring per-se, but still sub-optimal. Since build package knows how to use compiled sqlite3, there should not be need for rebuilding sqlite3 from the source just for the sake of tests.

Upstream README can be misleading. Many people seems to write them like package manages do not exist.

Reply:
I'm using `buildPythonPackage`. @SuperSandro2000 @KAction I applied the required changes but I still get the same error.